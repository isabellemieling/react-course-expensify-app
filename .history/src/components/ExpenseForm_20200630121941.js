import React from 'react';

export default class ExpenseForm extends React.Component {
    render() {
        return (
            <div>
                <form>
                    <input
                        type="text"
                        placeholder="Description"
                        autoFocus
                    />
                </form>
            </div>
        )
    }
}