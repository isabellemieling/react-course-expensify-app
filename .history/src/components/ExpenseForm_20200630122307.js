import React from 'react';

export default class ExpenseForm extends React.Component {
    state = {
        description: ''
    }
    render() {
        return (
            <div>
                <form>
                    <input
                        type="text"
                        placeholder="Description"
                        autoFocus
                    />
                    <input
                        type="number"
                        placeholder="Amount"
                    />
                    <textarea
                        placeholder="Add a note for your expense (optional)"
                    >
                    </textarea>
                    <button>Add Expense</button>
                </form>
            </div>
        )
    }
}