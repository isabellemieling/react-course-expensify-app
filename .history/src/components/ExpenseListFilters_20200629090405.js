import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter } from '../actions/filters';

const ExpenseListFilters = (props) => (
    <div>
        <input type="text" defaultValue={props.filters.text} onChange={(e) => {
            props.dispatch(setTextFilter(e.target.value));
        }} />
        <select>
            <option>Date</option>
            <option>Amount</option>
        </select>
    </div>
);

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    };
};

export default connect(mapStateToProps)(ExpenseListFilters); k