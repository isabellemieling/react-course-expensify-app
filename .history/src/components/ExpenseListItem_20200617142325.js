import React from 'react'; 
import { connect } from 'react-redux';
import { removeExpense } from '../actions/expenses';

// Below we have to destructure the props 
const ExpenseListItem = ( {dispatch, description, amount, createdAt, id} ) => (
    <div>
        <p>Item: {description}</p>
        <p>Cost: {amount}</p>
        <p>Date added: {createdAt}</p>
        <button onClick={() => {
            dispatch(removeExpense({ id }));
            }}>Remove</button>
    </div>
);


export default connect()(ExpenseListItem); 

