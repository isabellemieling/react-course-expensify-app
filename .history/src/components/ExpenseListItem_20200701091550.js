import React from 'react'; 
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { removeExpense } from '../actions/expenses';

// Below we have to destructure the props 
const ExpenseListItem = ( {dispatch, description, amount, createdAt, id} ) => (
    <div>
        <Link to={`/edit/${id}`}>
            <p>Item: {description}</p>
        </Link>
        <p>Cost: {amount}</p>
        <p>Date added: {createdAt}</p>
        <button onClick={() => {
            dispatch(removeExpense({ id }));
            }}>Remove</button>
    </div>
);


export default connect()(ExpenseListItem); 

