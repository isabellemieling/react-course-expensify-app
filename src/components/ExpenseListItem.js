import React from 'react'; 
import { Link } from 'react-router-dom';

// Below we have to destructure the props 
const ExpenseListItem = ( { description, amount, createdAt, id} ) => (
    <div>
        <Link to={`/edit/${id}`}>
            <p>Item: {description}</p>
        </Link>
        <p>Cost: {amount}</p>
        <p>Date added: {createdAt}</p>
    </div>
);


export default ExpenseListItem; 

