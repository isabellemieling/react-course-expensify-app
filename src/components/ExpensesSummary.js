import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total.js';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');

    return (
        <div>
            <p>Viewing {expenseCount} {expenseWord} totally {formattedExpensesTotal} </p>
        </div>
    );
};

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);



