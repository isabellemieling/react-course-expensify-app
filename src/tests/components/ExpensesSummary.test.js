import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({
    adapter: new Adapter()
});


test('Should correctly render expenses summary with one expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={234} />);
    expect(wrapper).toMatchSnapshot();
});

test('Should correctly render expenses summary with multiple expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={12} expensesTotal={22234} />);
    expect(wrapper).toMatchSnapshot();
});