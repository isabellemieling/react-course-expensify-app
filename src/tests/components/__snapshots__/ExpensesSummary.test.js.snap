// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render expenses summary with multiple expense 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpensesSummary
    expenseCount={12}
    expensesTotal={22234}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p>
        Viewing 
        12
         
        expenses
         totally 
        $222.34
         
</p>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Viewing ",
          12,
          " ",
          "expenses",
          " totally ",
          "$222.34",
          " ",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Viewing ",
        12,
        " ",
        "expenses",
        " totally ",
        "$222.34",
        " ",
      ],
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          Viewing 
          12
           
          expenses
           totally 
          $222.34
           
</p>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Viewing ",
            12,
            " ",
            "expenses",
            " totally ",
            "$222.34",
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Viewing ",
          12,
          " ",
          "expenses",
          " totally ",
          "$222.34",
          " ",
        ],
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Should correctly render expenses summary with one expense 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpensesSummary
    expenseCount={1}
    expensesTotal={234}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p>
        Viewing 
        1
         
        expense
         totally 
        $2.34
         
</p>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Viewing ",
          1,
          " ",
          "expense",
          " totally ",
          "$2.34",
          " ",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Viewing ",
        1,
        " ",
        "expense",
        " totally ",
        "$2.34",
        " ",
      ],
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          Viewing 
          1
           
          expense
           totally 
          $2.34
           
</p>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Viewing ",
            1,
            " ",
            "expense",
            " totally ",
            "$2.34",
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Viewing ",
          1,
          " ",
          "expense",
          " totally ",
          "$2.34",
          " ",
        ],
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
